CDCgov/prime-reportstream
# 6102 Convert non-repeating HL7 segments from FHIR

This PR adds the capability to convert FHIR to HL7.  Right now only non-repeating segments are supported.  I also added a PRIME CLI command to convert FHIR to HL7.

Test Steps:
1. Check out the branch for this PR.
2. Verify the tests pass by running the command `./gradlew package` and verifying the build is successful.
3. Run the following command to convert a FHIR bundle to HL7.  Note the conversion at this point is minimal, but seeing that an HL7 message is generated is the point.  
`./prime fhirdata --input-file "src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir" -s "src/test/resources/fhirengine/translation/hl7/schema/schema-read-test-01/ORU_R01.yml"`
4. Verify an HL7 message is generated from the conversion.  You will see an output like this:
```
MSH|^~\&||||||||d231207a-9456-43b3-8a26-9cb48c0b0525
PID|1||||BIERLY
```

## Changes
- Added FHIR to HL7 converter for non-repeating segments
- Added PRIME CLI command to convert FHIR to HL7

## Checklist

### Testing

### Process

## Linked Issues
- #6102 

## To Be Done


## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
